scipy/stats/tests/test_distributions.py

```diff
@@ -379,140 +379,204 @@ def test_pdf_r(self):
         #    function(x) GeneralizedHyperbolic::ghypMom(
-        #        order = x, lambda = 2, alpha = 2, beta = 1, delta = 1,
-        #        momType = 'raw')
+        #        order = x, lambda = 2, alpha = 2,
+        #        beta = 1, delta = 1.5, mu = 0.5
+        #        momType = 'mu')
         # )
-        
+
         vals_R = np.array([
             1.86848366948115, 6.35545100844345,
             26.8275110062306, 141.581144754983
-
             ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
         mvsk = stats.genhyperbolic(
-            lmbda=2,
-            alpha=2,
-            beta=1,
-            delta=1.5,
-            loc = 0
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
             ).stats('mvsk')
+
+        mvsk = list(mvsk)
+
+        # center and de-standardize moments to match R results
+        mvsk[0] -= mu
+        mvsk[2] *= np.float_power(delta, 3)
+        mvsk[3] *= np.float_power(delta, 4)
+
         assert_allclose(mvsk, vals_R, atol=1e-9, rtol=1e-8)
 
     def test_rvs(self):
         # KS test
-        #lmbda = 1, alpha = 1, beta = 0, delta = 1, mu = 0
-        gh = stats.genhyperbolic(lmbda=1, alpha=1, beta=0, delta=1, loc=0)
+        # p = 1, alpha = 1, beta = 0
+        gh = stats.genhyperbolic(p=1, a=1, b=0)
         _, p = stats.kstest(gh.rvs(size=1500, random_state=1234), gh.cdf)
+
         assert_equal(p > 0.05, True)
 
     def test_pdf_t(self):
-        # Test Against T-Student with 1 - 100 df
+        # Test Against T-Student with 1 - 30 df
         res = []
-        for df in np.linspace(1, 100, 20):
+        for df in np.linspace(1, 30, 20):
             # in principle alpha should be zero in practice for big lmbdas
             # alpha cannot be too small else pdf does not integrate
             alpha = np.float_power(df, 2)*np.finfo(np.float32).eps
+            beta = 0
+            mu = 0
+            delta = np.sqrt(df)
+            alpha_hat = alpha * delta
+            beta_hat = beta * delta
             gh = stats.genhyperbolic(
-                lmbda=-df/2,
-                alpha=alpha,
-                beta=0,
-                delta=np.sqrt(df)
+                p=-df/2,
+                a=alpha_hat,
+                b=beta_hat,
+                loc=mu,
+                scale=delta
                 )
             x = np.linspace(gh.ppf(0.01), gh.ppf(0.99), 100)
             res.append(assert_allclose(
                 gh.pdf(x), stats.t.pdf(x, df),
                 atol=1e-8)
                 )
+
         np.all(res)
 
     def test_pdf_laplace(self):
-        # Test Against Laplace with location param 1-100
+        # Test Against Laplace with location param [-10, 10]
         res = []
-        for loc in np.linspace(-100, 100, 20):
+        for loc in np.linspace(-10, 10, 20):
```

Comment:
Now that you've vectorized things, you don't need the for loops in tests, right?

Reply:
I am not sure i can have both an array of parameters (e.g. `loc=np.linspace(-10, 10, 20)`) and still be able to generate multiple points in the pdf. at least scipy does not seem to allow to do both things at the same time.

```
gh = stats.genhyperbolic(1, 1, 0, np.linspace(-1, 1, 3))
gh.pdf(np.linspace(-10, 10, 100))
# this code throws the following ValueError: operands could not be broadcast together with shapes (100,) (3,)
```

Reply:
The arrays need to be along different axes in order to indicate that you want all of those PDF values for all of those parameter values.
For instance, `np.linspace(-1, 1, 3)` could be `np.linspace(-1, 1, 3)[:, np.newaxis]` to make it a column instead of a row. Although, I would recommend that you write it as [-1, 0, 1] and instead change the PDF to a column:
`np.linspace(-10, 10, 100)[:, np.newaxis]`

This will generate a 100 x 3 matrix - the pdf at each point for each of the values of the parameter.

You can do this with any number of perpendicular dimensions. You might look into other [array manipulation routines](https://numpy.org/doc/stable/reference/routines.array-manipulation.html) like `reshape` if you need many more dimensions.